src/Config/ConfigurationStore.php

```diff
@@ -2,170 +2,50 @@
 
 namespace Eppo\Config;
 
-use Eppo\Bandits\BanditReferenceIndexer;
-use Eppo\Bandits\IBanditReferenceIndexer;
-use Eppo\Cache\CacheType;
-use Eppo\Cache\NamespaceCache;
-use Eppo\DTO\Bandit\Bandit;
-use Eppo\DTO\Flag;
-use Eppo\Exception\EppoClientException;
-use Eppo\Exception\InvalidArgumentException;
-use Eppo\Exception\InvalidConfigurationException;
-use Eppo\Validator;
+use Eppo\DTO\ConfigurationWire\ConfigurationWire;
 use Psr\SimpleCache\CacheInterface;
+use Psr\SimpleCache\InvalidArgumentException;
 
-class ConfigurationStore implements IConfigurationStore
+class ConfigurationStore
 {
-    private CacheInterface $flagCache;
-    private CacheInterface $banditCache;
-    private CacheInterface $metadataCache;
+    private const CONFIG_KEY = "EPPO_configuration_v1";
+    private ?Configuration $configuration = null;
 
-    // Key for storing bandit variations in the metadata cache.
-    private const BANDIT_VARIATION_KEY = 'banditVariations';
-
-    /**
-     * @param CacheInterface $cache
-     */
-    public function __construct(CacheInterface $cache)
+    public function __construct(private readonly CacheInterface $cache)
     {
-        $this->flagCache = new NamespaceCache(CacheType::FLAG, $cache);
-        $this->banditCache = new NamespaceCache(CacheType::BANDIT, $cache);
-        $this->metadataCache = new NamespaceCache(CacheType::META, $cache);
     }
 
-    /**
-     * @param string $key
-     * @return Flag|null
-     */
-    public function getFlag(string $key): ?Flag
+    public function getConfiguration(): Configuration
     {
-        try {
-            $result = $this->flagCache->get($key);
-            if ($result == null) {
-                return null;
-            }
-
-            $inflated = $result;
-            return $inflated === false ? null : $inflated;
-        } catch (\Psr\SimpleCache\InvalidArgumentException $e) {
-            // Simple cache throws exceptions when a keystring is not a legal value (characters {}()/@: are illegal)
-            syslog(LOG_WARNING, "[EPPO SDK] Illegal flag key ${key}: " . $e->getMessage());
-            return null;
-        }
-    }
+        if ($this->configuration === null) {
+            try {
+                $cachedConfig = $this->cache->get(self::CONFIG_KEY);
+                if (!$cachedConfig) {
+                    return Configuration::emptyConfig(); // Empty config
+                }
 
-    /**
-     * @param array $flags
-     * @param IBanditReferenceIndexer|null $banditVariations
-     * @throws EppoClientException
-     */
-    public function setUnifiedFlagConfiguration(array $flags, ?IBanditReferenceIndexer $banditVariations = null): void
-    {
-        try {
-            // Clear stored config before setting data.
-            $this->flagCache->clear();
+                $json = json_decode($cachedConfig, true);
```

Comment:
is `$json` the best name for this variable? Isn't it now an associative array as opposed to a JSON string?

That being said, I can't think of a better name. `$decodedJson` maybe? 😛  In other places we seem to use `$arr` to represent the same thing.

Or is `$json` some PHP convention?

Reply:
Nah you're right, $arr is most phpish.